BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/manager/ConnectionManager.java

```diff
@@ -79,17 +86,48 @@ public boolean teleportPlayerToServer(Player p, String server, PlayerChangeServe
 			BetterShardsPlugin.getInstance().getLogger().log(Level.INFO, "During BetterShards teleport, removing player {0} from vehicle", p.getUniqueId());
 			p.getVehicle().eject();
 		}
-		if (p.isDead()) {
+		if (p.isDead() && !delayedTransit.contains(p.getUniqueId())) {
 			throw new PlayerStillDeadException();
 		}
 		transitManager.addPlayerToExitTransit(p.getUniqueId(), server); // So the player isn't tried to be sent twice.
 		MercuryManager.warnOfArrival(p.getUniqueId(), server); //so target server prepares
 		CustomWorldNBTStorage st = CustomWorldNBTStorage.getWorldNBTStorage();
 		st.save(p, st.getInvIdentifier(p.getUniqueId()), true);
+		
+		p.setGameMode(GameMode.SPECTATOR); // So they cant do anything.
```

Comment:
We could add a callback to the async save, and run setgamemode after the async save is complete to address your concern @Maxopoly 

It's a worthy consideration. The lock is grabbed synchronously, so no further saves will run while the lock is held, but if the in-memory representation of inventory was swapped out by GM3 to be empty or whatever, that in-memory representation is what would be asynchronously saved.